code/datums/skills/craft.dm

```diff
@@ -10,27 +10,163 @@
+		if(SKILL_LEVEL_NONE)
+			return 1
+		if(SKILL_LEVEL_NOVICE)
+			return 1
+		if(SKILL_LEVEL_APPRENTICE)
+			return 1
+		if(SKILL_LEVEL_JOURNEYMAN)
+			return 0.5
+		if(SKILL_LEVEL_EXPERT)
+			return 0.33
+		if(SKILL_LEVEL_MASTER)
+			return 0.25
+		if(SKILL_LEVEL_LEGENDARY)
+			return 0.1
+
 /datum/skill/craft/smelting
 	name = "Smelting"
 
+/datum/skill/craft/smelting/get_skill_speed_modifier(level)
+	switch(level)
+		if(SKILL_LEVEL_NONE)
+			return 1
+		if(SKILL_LEVEL_NOVICE)
+			return 1
+		if(SKILL_LEVEL_APPRENTICE)
+			return 1
+		if(SKILL_LEVEL_JOURNEYMAN)
+			return 0.5
+		if(SKILL_LEVEL_EXPERT)
+			return 0.33
+		if(SKILL_LEVEL_MASTER)
+			return 0.25
+		if(SKILL_LEVEL_LEGENDARY)
+			return 0.1
+
 /datum/skill/craft/carpentry
 	name = "Carpentry"
 
+/datum/skill/craft/carpentry/get_skill_speed_modifier(level)
+	switch(level)
+		if(SKILL_LEVEL_NONE)
+			return 1
+		if(SKILL_LEVEL_NOVICE)
+			return 1
+		if(SKILL_LEVEL_APPRENTICE)
+			return 1
+		if(SKILL_LEVEL_JOURNEYMAN)
+			return 0.5
+		if(SKILL_LEVEL_EXPERT)
+			return 0.33
+		if(SKILL_LEVEL_MASTER)
+			return 0.25
+		if(SKILL_LEVEL_LEGENDARY)
+			return 0.1
+
 /datum/skill/craft/masonry
 	name = "Masonry"
 
+/datum/skill/craft/masonry/get_skill_speed_modifier(level)
+	switch(level)
+		if(SKILL_LEVEL_NONE)
+			return 1
+		if(SKILL_LEVEL_NOVICE)
+			return 1
+		if(SKILL_LEVEL_APPRENTICE)
+			return 1
+		if(SKILL_LEVEL_JOURNEYMAN)
+			return 0.5
+		if(SKILL_LEVEL_EXPERT)
+			return 0.33
+		if(SKILL_LEVEL_MASTER)
+			return 0.25
+		if(SKILL_LEVEL_LEGENDARY)
+			return 0.1
+
 /datum/skill/craft/cooking
 	name = "Cooking"
 
+/datum/skill/craft/cooking/get_skill_speed_modifier(level)
+	switch(level)
+		if(SKILL_LEVEL_NONE)
+			return 1
+		if(SKILL_LEVEL_NOVICE)
+			return 1
+		if(SKILL_LEVEL_APPRENTICE)
+			return 1
+		if(SKILL_LEVEL_JOURNEYMAN)
+			return 0.5
+		if(SKILL_LEVEL_EXPERT)
+			return 0.33
+		if(SKILL_LEVEL_MASTER)
+			return 0.25
+		if(SKILL_LEVEL_LEGENDARY)
+			return 0.1
+
 /datum/skill/craft/engineering
 	name = "Engineering"
 
+/datum/skill/craft/engineering/get_skill_speed_modifier(level)
+	switch(level)
+		if(SKILL_LEVEL_NONE)
+			return 1
+		if(SKILL_LEVEL_NOVICE)
+			return 1
+		if(SKILL_LEVEL_APPRENTICE)
+			return 1
+		if(SKILL_LEVEL_JOURNEYMAN)
+			return 0.5
+		if(SKILL_LEVEL_EXPERT)
+			return 0.33
+		if(SKILL_LEVEL_MASTER)
+			return 0.25
+		if(SKILL_LEVEL_LEGENDARY)
+			return 0.1
+
 /datum/skill/craft/hunting
 	name = "Hunting"
 	desc = ""
 	dreams = list(
 		"...you feel the forest's embrace as you drift into dreams, visions of stalking prey and the thrill of the hunt filling your mind.."
 	)
+
+/datum/skill/craft/hunting/get_skill_speed_modifier(level)
+	switch(level)
+		if(SKILL_LEVEL_NONE)
+			return 1
+		if(SKILL_LEVEL_NOVICE)
+			return 1
+		if(SKILL_LEVEL_APPRENTICE)
+			return 1
+		if(SKILL_LEVEL_JOURNEYMAN)
+			return 0.5
+		if(SKILL_LEVEL_EXPERT)
+			return 0.33
+		if(SKILL_LEVEL_MASTER)
+			return 0.25
+		if(SKILL_LEVEL_LEGENDARY)
+			return 0.1
```

Comment:
can you ever achieve this even as the assigned role at 0.1? Most start from master or so iirc.

Reply:
You misread the code. The 0.1 (10%) skill gain rate is when you already have legendary. And even if it was 0.1, you'd easily achieve this from just doing your work as a master. It's not hard to gain skills at all. This is good, read the code proper next time.

Reply:
right, either way its already slow unless you spam make items for 2 hours, i dont know the purpose of making it even worse.

Reply:
To give professional classes purpose, otherwise you can just remove them and make the whole server nothing but generic adventurers.